components/feature/night-mode/src/main/java/org/mozilla/rocket/nightmode/themed/NightTheme.kt

```diff
@@ -0,0 +1,28 @@
+package org.mozilla.rocket.nightmode.themed
+
+interface NightTheme : DefaultTheme {
+
+    fun setNightMode(isNight: Boolean) {
+        if (isNightTheme() != isNight) {
+            if (isNight) {
+                addThemeState(ThemedWidgetUtils.ThemeState.NIGHT)
+            } else {
+                removeThemeState(ThemedWidgetUtils.ThemeState.NIGHT)
+            }
+            notifyRefreshDrawableState()
+        }
+    }
+
+    fun isNightTheme(): Boolean {
+        return themeState and ThemedWidgetUtils.ThemeState.NIGHT.flag != 0
+    }
+
+    override fun getThemeStateSize(): Int {
```

Comment:
I'm not sure if we really need to expose this. Maybe just let outside users use getThemeDrawableState().size?

Reply:
Good point. I haven't noticed that!